libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/alert/IncidentAlert.kt

```diff
@@ -0,0 +1,251 @@
+        if (subType != other.subType) return false
+        if (subTypeDescription != other.subTypeDescription) return false
+        if (alertcCodes != other.alertcCodes) return false
+
+        return true
+    }
+
+    /**
+     * Returns a hash code value for the object.
+     */
+    override fun hashCode(): Int {
+        var result = super.hashCode()
+        result = 31 * result + incidentType.hashCode()
+        result = 31 * result + incidentId.hashCode()
+        result = 31 * result + incidentCongestion.hashCode()
+        result = 31 * result + isClosed.hashCode()
+        result = 31 * result + creationTime.hashCode()
+        result = 31 * result + startTime.hashCode()
+        result = 31 * result + endTime.hashCode()
+        result = 31 * result + description.hashCode()
+        result = 31 * result + subType.hashCode()
+        result = 31 * result + subTypeDescription.hashCode()
+        alertcCodes?.forEach {
+            result = 31 * result + it.hashCode()
+        }
+        return result
+    }
+
+    /**
+     * Returns a string representation of the object.
+     */
+    override fun toString(): String {
+        return "IncidentAlert(" +
+            "incidentType=$incidentType" +
+            "incidentId=$incidentId" +
+            "incidentId=$incidentId" +
+            "incidentCongestion=$incidentCongestion" +
+            "isClosed=$isClosed" +
+            "creationTime=$creationTime" +
+            "startTime=$startTime" +
+            "endTime=$endTime" +
+            "description=$description" +
+            "subType=$subType" +
+            "subTypeDescription=$subTypeDescription" +
+            "alertcCodes=$alertcCodes" +
+            "), ${super.toString()}"
+    }
+
+    /**
+     * Use to create a new instance.
+     *
+     * @see IncidentAlert
+     */
+    class Builder(
+        private val coordinate: Point,
+        private val distance: Double
+    ) {
+        private var alertGeometry: RouteAlertGeometry? = null
+        private var incidentType: Int = IncidentAlertType.INCIDENT_UNKNOWN
+        private var incidentId: String? = null
+        @IncidentAlertImpact.Impact private var impact: String? = null
+        private var incidentCongestion: IncidentCongestion? = null
+        private var isClosed: Boolean = false
+        private var creationTime: Date? = null
+        private var startTime: Date? = null
+        private var endTime: Date? = null
+        private var description: String? = null
+        private var subType: String? = null
+        private var subTypeDescription: String? = null
+        private var alertcCodes: List<Int>? = null
+
+        /**
+         * Add optional geometry if the alert has a length.
+         */
+        fun alertGeometry(alertGeometry: RouteAlertGeometry?): Builder = apply {
+            this.alertGeometry = alertGeometry
+        }
+
+        /**
+         * Unique identifier for incident. Must be **non-null**.
```

Comment:
Should it be required in the builder constructor in this case?

Reply:
I thought about it, and one more question that appear is `should we allow change id, so should it be mutable?`. I would say yes in both cases

Reply:
Why would it need to be mutable, do you have a use-case in mind? We should create a new alert object if it has to have a different ID.

Reply:
I could imagine if we want to merge a few incidents in one. Only possible way is make the same id for them

Reply:
Just FYI: The intent of the id was to allow visual union of incidents on routeline with incidents on the map.

Reply:
thanks! From this perspective ids should be un-mutable